saleor/graphql/product/tests/queries/test_product_variants_query_with_where.py

```diff
@@ -172,3 +173,103 @@ def test_product_variant_filter_by_sku(
     assert len(variants) == len(indexes)
     skus = {node["node"]["sku"] for node in variants}
     assert skus == {product_variant_list[index].sku for index in indexes}
+
+
+@pytest.mark.parametrize(
+    "stocks_to_create",
+    [[], [("1", 0), ("2", 1), ("4", 1)]],
+)
+def test_product_variant_where_in_stock(
+    api_client,
+    product_variant_list,
+    channel_USD,
+    stocks_to_create,
+    shipping_zones_with_warehouses,
+):
+    # given
+    warehouse = shipping_zones_with_warehouses[0].warehouses.first()
+    channel_USD.warehouses.add(warehouse)
+    Stock.objects.bulk_create(
+        [
+            Stock(
+                product_variant=ProductVariant.objects.filter(sku=sku).first(),
+                quantity=quantity,
+                warehouse=warehouse,
+            )
+            for sku, quantity in stocks_to_create
+        ]
+    )
+    variant_ids = [
+        graphene.Node.to_global_id("ProductVariant", variant.pk)
+        for variant in product_variant_list
+    ]
+    variables = {
+        "channel": channel_USD.slug,
+        "where": {"stockAvailability": "IN_STOCK", "ids": variant_ids},
+    }
+
+    # when
+    response = api_client.post_graphql(PRODUCT_VARIANTS_WHERE_QUERY, variables)
+
+    # then
+    data = get_graphql_content(response)
+    variants = data["data"]["productVariants"]["edges"]
+
+    positive_stock_variants = [
+        sku for sku, quantity in stocks_to_create if quantity > 0
+    ]
+    assert [variant["node"]["sku"] for variant in variants] == positive_stock_variants
```

Comment:
I suggest comparing the values as sets to avoid flaky test behavior due to ordering differences.